clang/test/dpct/helper_files_ref/include/rng_utils.hpp

```diff
@@ -174,6 +176,235 @@ template <typename engine_t> class rng_generator {
+        dpct::cpu_device().default_queue(), _seed);
+    break;
+  case random_engine_type::mrg32k3a:
+    _mrg32k3a_engine = std::make_shared<oneapi::mkl::rng::mrg32k3a>(
+        dpct::cpu_device().default_queue(), _seed);
+    break;
+  case random_engine_type::mt2203:
+    _mt2203_engine = std::make_shared<oneapi::mkl::rng::mt2203>(
+        dpct::cpu_device().default_queue(), _seed);
+    break;
+  case random_engine_type::mt19937:
+    _mt19937_engine = std::make_shared<oneapi::mkl::rng::mt19937>(
+        dpct::cpu_device().default_queue(), _seed);
+    break;
+  case random_engine_type::sobol:
+    _sobol_engine = std::make_shared<oneapi::mkl::rng::sobol>(
+        dpct::cpu_device().default_queue(), _dimensions);
+    break;
+  }
+  return 0;
+}
+
+int set_generator_seed(const std::uint64_t &seed) {
+  if (seed == _seed) {
+    return 0;
+  }
+  destroy_generator();
+  switch (_type) {
+  case random_engine_type::philox4x32x10:
+    _philox4x32x10_engine = std::make_shared<oneapi::mkl::rng::philox4x32x10>(
+        dpct::cpu_device().default_queue(), seed);
+    break;
+  case random_engine_type::mrg32k3a:
+    _mrg32k3a_engine = std::make_shared<oneapi::mkl::rng::mrg32k3a>(
+        dpct::cpu_device().default_queue(), seed);
+    break;
+  case random_engine_type::mt2203:
+    _mt2203_engine = std::make_shared<oneapi::mkl::rng::mt2203>(
+        dpct::cpu_device().default_queue(), seed);
+    break;
+  case random_engine_type::mt19937:
+    _mt19937_engine = std::make_shared<oneapi::mkl::rng::mt19937>(
+        dpct::cpu_device().default_queue(), seed);
+    break;
+  default:
+    break;
+  }
+  return 0;
+}
+
+int set_generator_dimensions(const std::uint32_t &dimensions) {
+  if (dimensions == _dimensions) {
+    return 0;
+  }
+  destroy_generator();
+  switch (_type) {
+  case random_engine_type::sobol:
+    _sobol_engine = std::make_shared<oneapi::mkl::rng::sobol>(
+        dpct::cpu_device().default_queue(), dimensions);
+    break;
+  default:
+    break;
+  }
+  return 0;
+}
+
+int set_generator_queue(const sycl::queue &queue) {
+  if (queue == _queue) {
+    return 0;
+  }
+  destroy_generator();
+  switch (_type) {
+  case random_engine_type::philox4x32x10:
+    _philox4x32x10_engine =
+        std::make_shared<oneapi::mkl::rng::philox4x32x10>(queue, _seed);
+    break;
+  case random_engine_type::mrg32k3a:
+    _mrg32k3a_engine =
+        std::make_shared<oneapi::mkl::rng::mrg32k3a>(queue, _seed);
+    break;
+  case random_engine_type::mt2203:
+    _mt2203_engine = std::make_shared<oneapi::mkl::rng::mt2203>(queue, _seed);
+    break;
+  case random_engine_type::mt19937:
+    _mt19937_engine = std::make_shared<oneapi::mkl::rng::mt19937>(queue, _seed);
+    break;
+  case random_engine_type::sobol:
+    _sobol_engine =
+        std::make_shared<oneapi::mkl::rng::sobol>(queue, _dimensions);
+    break;
+  }
+  return 0;
+}
+
+int destroy_generator() {
```

Comment:
1. is it possible to use destructor?
2. shorter name anyway
3. return value has no use
```suggestion
void destroy() {
```

Reply:
1. MKL API not support change the rng's kind or seed, so we need create a new object when user change the kind or seed. I think provide a method but not destructor is more convinient.
2. Fixed.
3. Fixed.